controlplane/rosa/controllers/rosacontrolplane_controller_test.go

```diff
@@ -150,3 +173,204 @@ func TestUpdateOCMClusterSpec(t *testing.T) {
+		},
+		Status: rosacontrolplanev1.RosaControlPlaneStatus{
+			Ready: true,
+			ID:    "rosa-control-plane-1",
+			// Version: "4.15.20",
+			Conditions: clusterv1.Conditions{clusterv1.Condition{
+				Type:     "Paused",
+				Status:   "False",
+				Severity: "",
+				Reason:   "NotPaused",
+				Message:  "",
+			}},
+		},
+	}
+
+	ownerCluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "owner-cluster-1",
+			Namespace: ns.Name,
+			UID:       types.UID("owner-cluster-1"),
+		},
+		Spec: clusterv1.ClusterSpec{
+			ControlPlaneRef: &corev1.ObjectReference{
+				Name:       rosaControlPlane.Name,
+				Kind:       "ROSAControlPlane",
+				APIVersion: rosacontrolplanev1.GroupVersion.String(),
+			},
+		},
+	}
+
+	rosaControlPlane.OwnerReferences = []metav1.OwnerReference{
+		{
+			Name:       ownerCluster.Name,
+			UID:        ownerCluster.UID,
+			Kind:       "Cluster",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+	}
+
+	mockCtrl := gomock.NewController(t)
+	ctx := context.TODO()
+	ocmMock := mocks.NewMockOCMClient(mockCtrl)
+	stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+
+	getCallerIdentityResult := &sts.GetCallerIdentityOutput{Account: aws.String("foo"), Arn: aws.String("arn:aws:iam::123456789012:rosa/foo")}
+	stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Return(getCallerIdentityResult, nil).Times(1)
+
+	expect := func(m *mocks.MockOCMClientMockRecorder) {
+		m.ValidateHypershiftVersion(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (bool, error) {
+			return true, nil
+		}).Times(1)
+		m.GetCluster(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterKey string, creator *rosaaws.Creator) (*v1.Cluster, error) {
+			sts := (&v1.STSBuilder{}).OIDCEndpointURL("oidc.com/oidc1")
+			aws := v1.NewAWS().AuditLog(v1.NewAuditLog().RoleArn("arn:aws:iam::123456789012:role/AuditLogRole")).STS(sts)
+			console := (&v1.ClusterConsoleBuilder{}).URL("console.redhat.com/cluster-123")
+			status := (&v1.ClusterStatusBuilder{}).State(v1.ClusterStateError)
+			version := (&v1.VersionBuilder{}).RawID(rosaControlPlane.Spec.Version)
+			mockCluster, _ := v1.NewCluster().AWS(aws).ID("cluster-1").Version(version).Status(status).Console(console).
+				RegistryConfig(v1.NewClusterRegistryConfig().
+					AdditionalTrustedCa(map[string]string{"trusted-ca": "-----BEGIN CERTIFICATE----- testcert -----END CERTIFICATE-----"}).
+					AllowedRegistriesForImport(v1.NewRegistryLocation().
+						DomainName("registry1.com").
+						Insecure(false))).
+				Build()
+			return mockCluster, nil
+		}).Times(1)
+	}
+
+	expect(ocmMock.EXPECT())
+
+	objects := []client.Object{ownerCluster, rosaControlPlane, secret, identity}
+	for _, obj := range objects {
+		createObject(g, obj, ns.Name)
+	}
+
+	// Add conditions, can't do this duirng creation
+	cpPh, err := patch.NewHelper(rosaControlPlane, testEnv)
+	g.Expect(err).ShouldNot(HaveOccurred())
+	rosaControlPlane.Status = rosacontrolplanev1.RosaControlPlaneStatus{
```

Comment:
Not sure, why patching the status ? we should check the status after Reconcile. 
You may just patch the ID other fields should be expected after Reconcile

Reply:
We need to set conditions for `Reconcile` to end without an error.